docs/PORTING.md

```diff
@@ -0,0 +1,40 @@
+# Bringing EVE up on a new CPU architecture
+
+EVE currently support x86 and ARM Edge Nodes. It works best when hardware assisted virtualization is available, but it can run with reduced functionality on pretty much anything supported by the Linux kernel. Most of the time, bringing EVE up on a new hardware configuration requires [peripheral hardware enablement](HARDWARE-BRINGUP.md) but in rare cases it may actually require supporting a brand new CPU model. The later consists of the following steps:
+
+1. Prerequisites
+2. Porting Alpine packages EVE depends on to the new CPU
+3. Hooking up new CPU emulation environment to Makefile and GitHub workflows
+4. Porting firmware, low-level bootloader and GRUB on the new CPU
+5. Porting Linux kernel to the new CPU
+6. Porting user-space EVE containers to the new CPU
+
+## 1. Prerequisites
+
+First of all, we are assuming that GCC and/or LLVM toolchain and Golang toolchain already provide support for the target CPU. We are also assuming that cross-compilation for both is an option (if either of these assumptions is broken -- you will have to engage in a heavy development effort bringing these toolchain up to speed with the target architecture). We are also assuming that qemu already supports the target CPU in either system or user mode.
```

Comment:
> cross-compilation for both is an option

Is it necessary? If we have qemu user support (or even OS support), we can just emulate and run it such that the toolchain thinks it is running on the target arch, rather than cross-compiling for it?

Reply:
Yes, but that is only when Alpine itself is available -- right now it is not. Like not a single bit of Alpine exists for riscv64. I'm working on Alpine port to riscv64 separately, but this is also a good opportunity to document how you would bootstrap that on the EVE side.